src/components/TopNavBar.tsx

```diff
@@ -1,35 +1,43 @@
-import React from 'react'
-import { NavLink } from 'react-router-dom'
+import React, { useEffect, useState } from 'react'
+import { NavLink, withRouter, useLocation } from 'react-router-dom'
 import { signOut } from 'synapse-react-client/dist/utils/SynapseClient'
 import logo from '../assets/favicon.png'
 
+const links = [
+    // {
+    //     text: 'Products',
+    //     path: '/products'
+    // },
+    {
+        text: "Profile",
+        path: '/authenticated/myprofile'
+    },
+    {
+        text: "Account Settings",
+        path: '/authenticated/myaccount'
+    }
+]
+
 const TopNavBar: React.FunctionComponent = () => {
-    const links = [
-        {
-            text: 'Products',
-            path: '/products'
-        },
-        {
-            text: "Profile",
-            path: '/authenticated/myprofile'
-        },
-        {
-            text: "Account Settings",
-            path: '/authenticated/myaccount'
-        }
-    ]
+    const [isVisible, setIsVisible] = useState(true)
+    const location = useLocation()
 
+    useEffect(() => {
+        setIsVisible(!!links.find(el => el.path === location.pathname))
+      }, [location])
+    
+    
     return(
-    <React.Fragment>
-        <nav className='top-nav'> 
-            <NavLink to={'/'}><img src={logo} style={{height:'32px', paddingRight: '20px'}}/></NavLink>
+    <>
+        {isVisible && <nav className='top-nav'> 
+            <NavLink to={'/'}><img src={logo} alt="Sage Bionetworks logo" style={{height:'32px', paddingRight: '20px'}}/></NavLink>
             {links.map(el => 
-                <NavLink to={el.path} activeClassName='isSelected' className={`nav-link`} >{el.text}</NavLink>
+                <NavLink to={el.path} key={el.text} activeClassName='isSelected' className={`nav-link`} >{el.text}</NavLink>
```

Comment:
important to set the key to a unique value for every child when creating children like this

Reply:
I think this informs react whether or not to rerender a node when comparing to the virtual dom